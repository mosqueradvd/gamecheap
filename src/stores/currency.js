import { writable } from 'svelte/store';

export const mxnTousd = 22.5357
export const copTousd = 3578.4545