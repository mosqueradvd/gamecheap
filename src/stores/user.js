import { writable } from 'svelte/store'

export const currentUser = writable(null);
export const colombianFlag = writable(true)