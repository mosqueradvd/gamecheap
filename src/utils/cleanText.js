const CleanText = (text) => {
  text = text.replace(/<[^>]*>?/g, '');
  text = text.trim()
  return text
}

export default CleanText