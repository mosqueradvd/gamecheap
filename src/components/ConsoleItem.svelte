<script>
  import cleanText from '../utils/cleanText';
  import { mxnTousd, copTousd } from '../stores/currency'; 
  export let segment;
  export let image = 'PS.png';
  export let url = 'mock';
  export let id = 1;
  export let name = 'PlayStation 4';
  export let price = 1000;
  export let description = 'PlayStation 4 description and features';
  export let commerce = 'Amazon';
  export let id_type_product = 1;
  export let product_type = "playstation";
  export let id_ecommerce = 1;

 switch (id_type_product) {
    case 1:
      product_type = "nintendo";
      break;
    case 2:
      product_type = "xbox";
      break;
    case 3:
      product_type = "playstation";
      break;
    default:
      product_type = "default";
      break;
  }
  switch (id_ecommerce) {
    case 1:
      commerce = "MercadoLibre";
      break;
    case 2:
      commerce = "ColombiaGamer";
      break;
    case 3:
      commerce = "OLX";
      break;
    case 4:
      commerce = "GamePlanet";
      break;
    case 5:
      commerce = "Sears";
      break;
    case 6:
      commerce = "MixUp";
      break;
    default:
      commerce = "default";
      break;
  }
  price = parseFloat(price)
  const colombianPrice = new Intl.NumberFormat("en-US", {style: "currency", currency: "USD"}).format(price/copTousd)
  const mexicanPrice = new Intl.NumberFormat("en-US", {style: "currency", currency: "USD"}).format(price/mxnTousd)
  description = cleanText(description)
</script>

<style>
  .console__item {
    max-width: 350px;
    min-height: 350px;
    text-align: center;
    margin-right: 1.3rem;
    line-height: 1.2rem;
    border-radius: 7px;
    box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.3);
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    padding: 20px 20px 0px 20px;
    margin-bottom: 1.3rem;
  }
  .nintendo {
    background-image: linear-gradient(to top, #e50020, #df001f, #d9001e, #d3001d, #cd001c);
    color: #fff;
  }
  .xbox {
    background-image: linear-gradient(to top, #107c10, #0f730f, #0d690e, #0c600c, #0b570b);
    color: #fff;
  }
  .playstation {
     background-image: linear-gradient(to top, #007aff, #006fe5, #0064cc, #035ab3, #0b4f9a);
    color: #fff;
  }
  .item__provider {
    color: #000;
    background-color: #cccc2e;
    font-weight: 600;
    margin: 10px auto 0;
    padding: 10px;
    border-radius: 6px;
  }
  h1 {
    font-size: var(--card-title);
  }

  h2 {
    font-size: var(--card-price);
  }

  p {
    font-size: var(--card-description);
    margin: 0;
  }

  img {
    width: 130px;
    margin: 0.5em;
  }
  a {
    text-decoration: none;
  }
  @media screen and (min-width: 700px) {
    h1 {
      font-size: 20px;
    }

    h2 {
      font-size: 18px;
    }

    p {
      font-size: 14px;
      margin: 0;
    }
  }
</style>

<div class={`console__item ${product_type}`} data-url={url} id={id} >
  <a aria-current={segment === 'item' ? 'page' : undefined} href={`item/${id}`}>
    <img src={image} alt="Item Imagen" />
    <h1>{name.substring(0, 20)}</h1>
    {#if id_ecommerce < 4}
      <h2>USD {colombianPrice}</h2>
      {:else}
      <h2>USD {mexicanPrice}</h2>
    {/if}
    <p class="description">{description.substring(0, 35)}</p>
  </a>
  <div class="item__provider">
    <a href={url} target="_blank">{`Ir a: ${commerce}`}</a>
  </div>
</div>
