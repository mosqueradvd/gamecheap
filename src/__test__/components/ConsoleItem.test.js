import { render as stlRender } from '@testing-library/svelte';
import '@testing-library/jest-dom/extend-expect';
import ConsoleItem from '../../components/ConsoleItem.svelte';
import playstationMock from './__mocks__/playstation_mock.json'
import cleanText from '../../utils/cleanText';
import { mxnTousd } from '../../stores/currency';

describe('<ConsoleItem /> Component', () => {  
  test('Check the UI of the ConsoleItem component', () => {
    const consoleItem = stlRender(ConsoleItem)
  
    expect(consoleItem).toMatchSnapshot()
  })

  test('Check proper render of properties', () => {
    const consoleItem = stlRender(ConsoleItem, {
      ...playstationMock, segment: 'item'
    })

    const title = consoleItem.container.getElementsByTagName('h1')[0]
    expect(title).toHaveTextContent(playstationMock.name)

    const price = consoleItem.container.getElementsByTagName('h2')[0]
    const expectedPrice = new Intl.NumberFormat(
      "en-US", 
      {style: "currency", currency: "USD"}
    )
    .format(playstationMock.price/mxnTousd)
    expect(price).toHaveTextContent(expectedPrice)

    const description = consoleItem.container.getElementsByClassName('description')[0]
    expect(description.textContent).toEqual(cleanText(playstationMock.description))

    const ecommerce = consoleItem.container
    .getElementsByClassName('item__provider')[0].children[0].textContent
    expect(ecommerce).toEqual('Ir a: GamePlanet')

    const classList = consoleItem.container.classList.contains('playstation')
    expect(classList).toBeTruthy
  })

  test('Check that cleanText() function works properly', () =>{
    const descriptionUncleaned = "<div>Esta consola es reacondicionada. \n\n\n    </div>\n";
    const descriptionCleaned = "Esta consola es reacondicionada.";
    const text = cleanText(descriptionUncleaned)
    expect(text).toEqual(descriptionCleaned)
  })
})