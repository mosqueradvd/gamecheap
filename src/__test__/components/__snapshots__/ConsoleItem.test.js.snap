// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConsoleItem /> Component Check the UI of the ConsoleItem component 1`] = `
Object {
  "component": ConsoleItem {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        "PlayStation 4 description and features",
        "MercadoLibre",
        "nintendo",
        undefined,
        "PS.png",
        "mock",
        1,
        "PlayStation 4",
        1,
        "$0.28",
        "$44.37",
        1000,
        1,
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
      ],
      "on_mount": Array [],
      "props": Object {
        "commerce": 1,
        "description": 0,
        "id": 6,
        "id_ecommerce": 8,
        "id_type_product": 12,
        "image": 4,
        "name": 7,
        "price": 11,
        "product_type": 2,
        "segment": 3,
        "url": 5,
      },
      "update": [Function],
    },
    "$capture_state": [Function],
    "$inject_state": [Function],
    "$set": [Function],
  },
  "container": <body>
    <div>
      <div
        class="console__item nintendo"
        data-url="mock"
        id="1"
      >
        <a
          href="item/1"
        >
          <img
            alt="Item Imagen"
            src="PS.png"
          />
           
          <h1>
            PlayStation 4
          </h1>
           
          <h2>
            USD $0.28
          </h2>
           
          <p
            class="description"
          >
            PlayStation 4 description and featu
          </p>
        </a>
         
        <div
          class="item__provider"
        >
          <a
            href="mock"
            target="_blank"
          >
            Ir a: MercadoLibre
          </a>
        </div>
      </div>
    </div>
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
